source/octf/trace/parser/ParsedIoTraceEventQueue.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright(c) 2012-2018 Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause-Clear
+ */
+
+#ifndef SOURCE_OCTF_TRACE_PARSER_PARSEDIOTRACEEVENTQUEUE_H
+#define SOURCE_OCTF_TRACE_PARSER_PARSEDIOTRACEEVENTQUEUE_H
+
+#include <memory>
+#include <string>
+#include <thread>
+#include <octf/proto/parsedTrace.pb.h>
+#include <octf/trace/parser/ParsedIoTraceEventHandler.h>
+#include <octf/utils/Exception.h>
+#include <octf/utils/NonCopyable.h>
+
+namespace octf {
+
+/**
+ * This class is a queue of parsed IO trace events.
+ *
+ * It parses a trace and allows to access parsed IOs using queue interface.
+ *
+ * @throws Exception when accessing empty queue (calling front or pop)
```

Comment:
It seems like the exception is thrown when event parsing had an exception, not when accessing an empty queue